src/Application/Models/Broker.js

```diff
@@ -51,7 +56,7 @@ class Broker {
     const index = _.findIndex(this._assets, { id: assetId });
     if (index === -1) {
       // Return the original broker, but this is not good.
-      return new Broker(this._assets);
+      return new Broker(this._assets, this._areStatsVisible);
```

Comment:
why not just `return this;` ?  seems to be functionally equivalent but saving the re-construction